examples/classifcation/text_classification.exs

```diff
@@ -0,0 +1,273 @@
+  end
+
+  defp decode(dictionary, word) do
+    with {:ok, decoded} <- Map.fetch(dictionary, word) do
+      decoded
+    else
+      _-> "Unknown"
+    end
+  end
+
+  def transform_words(word_labels, word_to_idx, label_to_idx, wcount, lcount) do
+
+    train_data =
+      word_labels
+      |> Enum.map(fn {word, label} ->
+        encode(word_to_idx, word)
+      end)
+      |> Enum.chunk_every(@sequence_length, 1, :discard)
+      |> Enum.drop(-1)
+      |> Nx.tensor
+      |> Nx.divide(wcount)
+      |> Nx.reshape({:auto, @sequence_length, 1})
+      |> Nx.to_batched_list(@batch_size)
+
+
+    train_labels =
+      word_labels
+      |> Enum.map(fn {word, label} ->
+          encode(label_to_idx, label)
+      end)
+      |> Enum.chunk_every(@sequence_length, 1, :discard)
+      |> Enum.drop(-1)
+      |> Nx.tensor
+      |> Nx.reshape({:auto, @sequence_length, 1})
+      |> Nx.to_batched_list(@batch_size)
+
+
+      {train_data, train_labels}
+  end
+
+
+  def predict(init_sequence, params, model, idx_to_char, idx_to_label, count, real) do
+
+        Enum.reduce(1..10, [], fn _, seq ->
+
+          p = Axon.predict(model, params, init_sequence)
+
+          predicted =
+            p
+            |> Nx.argmax(axis: -1)
+            |> Nx.to_flat_list
+            |> Enum.map(fn predict_sequence ->
+                decode(idx_to_label, predict_sequence)
+              end)
+
+
+          given =
+            init_sequence
+            |> Nx.to_flat_list
+            |> Enum.map(fn inpseq ->
+                {actual_result, _} =
+                  inpseq*count
+                  |> Float.floor
+                  |> Float.to_string
+                  |> Integer.parse
+
+                decode(idx_to_char, actual_result)
+              end)
+
+          rl =
+            real
+            |> Nx.to_flat_list
+            |> Enum.map(fn real_sequence ->
+                decode(idx_to_label, real_sequence)
+              end)
+
+          Enum.zip([given, predicted, rl])
+
+        end)
+        |> Enum.map(fn {input, output, r} ->
+                true? = if output == r do "CORRECT" else "INCORRECT" end
+
+                if true? == "CORRECT" do
+                  Logger.warn "Word was: #{input}, Predicted Category was: #{output} Real Result: #{r}, RETURNED: #{true?}"
+                else
+                  Logger.error "Word was: #{input}, Predicted Category was: #{output} Real Result: #{r}, RETURNED: #{true?}"
+                end
+            end)
+  end
+
+  defp from_file(filename, opts \\ []) do
+
+    data_stream =
+      filename
+      |> File.stream!()
+  end
+
+  def pre_process(streamed_data) do
+
+    char_label_data =
+    streamed_data
+    |> Stream.map(fn line ->
+      [token | entities] = String.split(line, @def_split)
+      if(
+      !is_nil(token) &&
+      token !== "" &&
+      String.length(token) !== 1 &&
+      line !== "\n" &&
+      token !== @start &&
+      token !== @ending &&
+      token not in @incorrect)
```

Comment:
```suggestion
      if token not in ["", @start, @ending] and token not in @incorrect and String.length(token) !== 1
```

I don't think `token` can ever be `nil` :)

Reply:
Just validated this & also changed some things based on suggestion. The reason for it there was the weird early handling of the row:
`"WORD LABEL LABEL LABEL\n"` when I adjusted the preprocessing as you suggested below it was much more obvious.